h2/src/main/org/h2/mvstore/SingleFileStore.java

```diff
@@ -231,20 +231,29 @@ protected long getAfterLastBlock_() {
 
     @Override
     public void backup(ZipOutputStream out) throws IOException {
-        boolean before = isSpaceReused();
-        setReuseSpace(false);
+        String f = correctFileName(FilePath.get(getFileName()).toRealPath().getName());
+        out.putNextEntry(new ZipEntry(f));
+        FileChannel in = originalFileChannel != null ? originalFileChannel : fileChannel;
+        MVStore mvStore = getMvStore();
+        boolean before = mvStore.setReuseSpace(false);
         try {
-            backupFile(out, getFileName(), originalFileChannel != null ? originalFileChannel : fileChannel);
+
+            long copied = IOUtils.copy(in, 0, out);
+
+            mvStore.executeFilestoreOperation(() -> {
+                try {
+                    IOUtils.copy(in, copied, out);
+                    writeCleanShutdownMark();
+                    IOUtils.copy(in, out, 0, 2 * BLOCK_SIZE);
```

Comment:
I don't think this is correct: The file header(s) will be copied at the tail of the backup file and not at the beginning.

Reply:
@vreuland Yes, you are right again, thank you! I don't know what I've been smoking. :smile_cat: 
Just don't have enough time to do it now.